packages/tx/src/transaction.ts

```diff
@@ -59,17 +59,19 @@ export default class Transaction {
 
     const [nonce, gasPrice, gasLimit, to, value, data, v, r, s] = values
 
+    const emptyBuffer = Buffer.from([])
+
     return new Transaction(
       {
         nonce: new BN(nonce),
         gasPrice: new BN(gasPrice),
         gasLimit: new BN(gasLimit),
         to: to && to.length > 0 ? new Address(to) : undefined,
         value: new BN(value),
-        data: data || Buffer.from([]),
-        v: v ? new BN(v) : undefined,
-        r: r ? new BN(r) : undefined,
-        s: s ? new BN(s) : undefined,
+        data: data ?? emptyBuffer,
+        v: !v?.equals(emptyBuffer) ? new BN(v) : undefined,
+        r: !r?.equals(emptyBuffer) ? new BN(r) : undefined,
+        s: !s?.equals(emptyBuffer) ? new BN(s) : undefined,
```

Comment:
I find these lines somewhat hard to read, replicated on REPL to trust them. No blocker though.

Reply:
Is the question mark necessary? What does it do? The `!` implies that it is defined, so I don't see what this `?` should do?

Reply:
I agree that this is not the easiest to read, but I found it the best option for this particular use. I am happy to change if there is a better suggestion.

> Is the question mark necessary? What does it do? The `!` implies that it is defined, so I don't see what this `?` should do?

The `!` negates the equals expression to ensure that it doesn't match `emptyBuffer`. It would imply that it is defined if it was in the position of the `?`. The `?` is optional chaining so that if it *is* `undefined`, it will revert to the `false` clause of the ternary and output `undefined`.

If ternary expressions could accept `&&` clauses, then It could be written as: `v && !v.equals(emptyBuffer) ? new BN(v) : undefined`

Reply:
Hi, the bug is actually not fixed (I should have noticed this earlier especially after your comment) - indeed, if `v` does not exist, the `?` operator makes it `false`, but it is then negated - so it is true. There's always a BN created.
```
      let r = undefined
      console.log(!r?.equals(emptyBuffer) ? new BN(r) : undefined)
```

Prints `<BN 0>`

Reply:
CC #1048

Reply:
hmmm oh man I am just testing this again, I have to check why the new tests I added in this PR are passing okay then...

I am also wondering why in the comment above I was confused because ternary expression *do* accept `&&` clauses (must have been confused with something else), so this example code I wrote might be more reasonable to read:
 
`v && !v.equals(emptyBuffer) ? new BN(v) : undefined`

I saw you updated in your PR to:

`r !== undefined && !r?.equals(emptyBuffer) ? new BN(r) : undefined,`

but in that case the optional chaining `?` is not needed anymore (since it's already being checked for undefined) so this is a lot more understandable:

`r !== undefined && !r.equals(emptyBuffer) ? new BN(r) : undefined,`

Reply:
Ah yes good point! I still find it very hard to read though, so if there's a better alternative that'd be great :)

Reply:
I opened a PR to fix this: https://github.com/ethereumjs/ethereumjs-monorepo/pull/1077